### PR TITLE
End-to-end Memory networks example for pytorch

### PR DESCRIPTION
**Pull request by uditsaxena**
This is an implementation of the end to end memory networks in pytorch from [End-To-End Memory Networks](http://arxiv.org/abs/1503.08895) using the Facebook [bAbI](http://arxiv.org/abs/1502.05698) dataset.

### CODE REVIEW

**Comment by frankxu2004** [comment]
I'm wondering what is the current status of this example?

**Comment by uditsaxena** [comment]
The PR is still open for this to be reviewed and merged.

**Comment by jekbradbury** [comment]
Can you explain why the temporal encoding parameters include batch size as a dimension? It's pretty strange in deep learning to have explicit batch-dependence like that other than through batchnorm.

**Comment by bkj** [comment]
FWIW -- this throws a bunch of warnings w/ `pytorch==0.4.0` and `python==3.6.5`
